### PR TITLE
Update the patch version to 4.25.1

### PR DESCRIPTION
**Pull request by v-smanchem**
### Issue describing the changes in this PR

Bumping up the patch version to 4.25.1

Cherry pick the commit https://github.com/Azure/azure-functions-host/commit/b42df4d4e63ab87f7a737f5a65dadf3eb7ba7de4

### Pull request checklist

* [x] My changes **do not** require documentation changes
    * [ ] Otherwise: Documentation issue linked to PR
* [ ] My changes **should not** be added to the release notes for the next release
    * [x] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
    * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] My changes **do not** require diagnostic events changes
    * Otherwise: I have added/updated all related diagnostic events and their documentation (Documentation issue linked to PR)
* [x] I have added all required tests (Unit tests, E2E tests)

### CODE REVIEW

File: release_notes.md [hunk 1]
```diff
@@ -3,7 +3,13 @@
 <!-- Please add your release notes in the following format:
 - My change description (#PR)
 -->
+<<<<<<< HEAD
```
Comment (VpOfEngineering) [inline release_notes.md]: This is an artifact from the merge commit - please carefully compare the difference between the release notes in the release branch and the release notes updates in Gavin's PR.

File: release_notes.md [hunk 2]
```diff
@@ -3,7 +3,13 @@
...
 - Update Python Worker Version to [4.16.0](https://github.com/Azure/azure-functions-python-worker/releases/tag/4.16.0)
+=======
+- Increase timeout for http proxying requests [#9433](https://github.com/Azure/azure-functions-host/pull/9433)
+
+- Update Python Worker Version to [4.14.0](https://github.com/Azure/azure-functions-python-worker/releases/tag/4.14.0)
+>>>>>>> b42df4d4e (Revert python worker to 4.14 (#9442))
```
Comment (VpOfEngineering) [inline release_notes.md]: here too

**Review by VpOfEngineering — CHANGES_REQUESTED** [review]

File: release_notes.md (hunk 1 above)
Comment (v-smanchem) [reply release_notes.md]: Updated the release notes

File: release_notes.md (hunk 2 above)
Comment (v-smanchem) [reply release_notes.md]: Updated the release notes

File: release_notes.md [hunk 3]
```diff
@@ -3,17 +3,10 @@
 <!-- Please add your release notes in the following format:
 - My change description (#PR)
 -->
-- Update Python Worker Version to [4.16.0](https://github.com/Azure/azure-functions-python-worker/releases/tag/4.16.0)
-- Update Java Worker Version to [2.12.1](https://github.com/Azure/azure-functions-java-worker/releases/tag/2.12.1)
```
Comment (VpOfEngineering) [inline release_notes.md:7]: As far as I can tell, the only line that should change is the Python worker line - downgrading it to 4.14.0. Can we make sure that is the only change in this PR?

**Review by VpOfEngineering — CHANGES_REQUESTED** [review]

**Comment by VpOfEngineering** [comment]
LGTM

**Review by VpOfEngineering — APPROVED** [review]

**Pull request merged**